elasticsearch/hooks/run

```diff
@@ -2,6 +2,9 @@
 
 exec 2>&1
 
+# Root needs to perform the below.
+sysctl -w vm.max_map_count={{cfg.sysctl.vm.max_map_count}}
```

Comment:
This probably belongs in the init hook, rather than the run hook. Same for the ulimit below. 

On a more philosophical level, I've got mixed feelings on this. On one hand, it feels like a perfectly natural thing to do. You're capturing the requirements of this application in its lifecycle hooks which I think is a good thing.  

The flip side is I worry that we're actually making this less portable.  I think of plan.sh not as a plan for Linux, but a plan for operating systems that Shell is the "native language".  Today this is just Linux, but this may not be the case in the future.  I don't want to block this because of what might be,  but I do think it can be a guide for what does and doesn't belong in the default plan.sh, and I think it might warrant a conversation about same.

Reply:
The ulimit likely has to be in the run hook since it is a per-process setting no?

Reply:
I've given a lot of thought to what goes in a init hook vs run hook and, to be honest, it's not clear.  There are definitely some things that will function in either hook and it's hard to discern which hook it is better to put it in.  Our [Best Practices Guide](https://www.habitat.sh/docs/best-practices/) does not cover much of this (other than to never call hab or sleep in a hook that is not a run hook).  

This feels like something that is definitely going to come up again as core plans continues to grow - @smacfarlane do you mind drafting an RFC around this to capture your thoughts on what should go in an init hook or run hook?  Let me know if you'd like to talk informally about this before submitting the formal RFC, I know it can be a little intimidating.